experimental/devices/pca9548/pca9548.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package pca9548
+
+import (
+	"errors"
+	"strconv"
+	"sync"
+
+	"periph.io/x/periph/conn"
+	"periph.io/x/periph/conn/i2c"
+	"periph.io/x/periph/conn/i2c/i2creg"
+	"periph.io/x/periph/conn/physic"
+)
+
+// DefaultOpts is the recommended default options.
+var DefaultOpts = Opts{Address: 0x70}
+
+// Opts is the pca9548 configuration.
+type Opts struct {
+	// Address pca9548 I²C Address. Valid addresses for the NXP pca9548 are 0x70
+	// to 0x77. The address is set by pulling A0~A2 low or high. Please refer
+	// to the datasheet.
+	Address int
+}
+
+// Dev is handle to a pca9548 I²C Multiplexer.
+type Dev struct {
+	// Immutable.
+	c        i2c.Bus
+	address  uint16
+	name     string
+	numPorts uint8
+
+	// Mutable.
+	mu         sync.Mutex
+	activePort uint8
+}
+
+// New creates a new handle to a pca9548 I²C multiplexer.
+func New(bus i2c.Bus, opts *Opts) (*Dev, error) {
+	if opts.Address < 0x70 || opts.Address > 0x77 {
+		return nil, errors.New("Address outside valid range of 0x70-0x77")
+	}
+	d := &Dev{
+		c:          bus,
+		activePort: 0xFF,
+		address:    uint16(opts.Address),
+		numPorts:   8,
+		name:       "pca9548-" + strconv.FormatUint(uint64(opts.Address), 16),
+	}
+	r := make([]byte, 1)
+	err := bus.Tx(uint16(opts.Address), nil, r)
+	if err != nil {
+		return nil, errors.New("could not establish communicate with multiplexer: " + err.Error())
+	}
+	return d, nil
+}
+
+// RegisterPorts registers multiplexer ports with the host. These ports can
+// then be used as any other i2c.Bus. Busses will be named "alias0", "alias1"
+// etc. If using more than one multiplexer note that the alias must be unique.
+func (d *Dev) RegisterPorts(alias string) error {
+	d.mu.Lock()
```

Comment:
I don't think this actually requires the lock, since it's only accessing immutable code. And it wouldn't be a big deal if a Tx() call would happen concurrently.